public/app/screens/main/sections/device-list/device-finder/device-finder.jsx

```diff
@@ -33,18 +54,22 @@ var DeviceFinder = React.createClass({
   },
 
   getDialog: function getDialog () {
-    switch (this.state.activity) {
-      case DeviceFinderActivity.SEARCHING:
-        return this.getSearchDialog();
-      case DeviceFinderActivity.NO_DEVICES:
-        return this.getNoResultsDialog();
-      default:
-        return null;
+    var dialogToRender = null; // no jsx markup dy default
+
+    if (this.state.activity === DeviceFinderActivity.SEARCHING) {
+      dialogToRender = this.getSearchDialog();
+    } else if (this.state.activity === DeviceFinderActivity.NO_DEVICES) {
+      dialogToRender = this.getNoResultsDialog();
+    } else if (this.state.activity === DeviceFinderActivity.DEVICE_FOUND) {
+      dialogToRender = this.getDeviceFoundDialog();
     }
```

Comment:
Why don't you use `switch` anymore? Is it best practise not to use switches in JS? You didn't do it for the other state as well (see comment before).